### DISCUSSION TITLE
How can I install Chirpity on Linux

### DISCUSSION THREAD

**Discussion by paulauvirage** [Mattk70/Chirpity-Electron#148] [Q&A] [answered]
Hello is there any plan to make this available for download on Linux (Ubuntu 20.04 LTS in my case)?

**Comment by Mattk70** [comment]
Hey, I've no plans to package a Linux app, but you should be able to get Chirpity running on Linux by following the instructions here: https://github.com/Mattk70/Chirpity-Electron/blob/master/README.md#running-the-application-from-source

**Comment by Mattk70** [comment]
@paulauvirage There is now an AppImage for Linux among the release packages

**Comment by paulauvirage** [comment]
Thank you  I've downloaded the AppImage

**Reply by gitalm** [reply]
Dear @Mattk70  and @paulauvirage

 i tried to run the AppImage on Ubuntu 24.04 and i got an errror.
`[660333:1223/074706.499428:FATAL:setuid_sandbox_host.cc(158)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /tmp/.mount_chirpinGaGu4/chrome-sandbox is owned by root and has mode 4755.
`
Thanks for your help. I couldn't change any permission in /tmp 
Thanks for the wonderful work. 
Johannes

**Reply by Mattk70** [reply, marked as answer]
Hi @gitalm, I recently upgraded to 24.04, and hadn't seen this issue until I tried to launch the AppImage from the terminal.

A quick Google leads me to believe the problem is due to Ubuntu's  introduction of  [restricted unprivileged user namespaces](https://ubuntu.com/blog/ubuntu-23-10-restricted-unprivileged-user-namespaces)

Nonetheless, did you try launching the AppImage from the Files program by double-clicking the file? This works fine for me.

If that doesn't work, your best bet will be to disable the Ubuntu feature, like this:

```
sudo sysctl -w kernel.apparmor_restrict_unprivileged_unconfined=0
sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
```

**Reply by gitalm** [reply]
Dear @Mattk70 ,

merry thanks 🎄🦅 for your help. 

1. I tried within the thunar explorer and *sudo dmesg* shows

```
audit: type=1400 audit(1734963607.919:301): apparmor="DENIED" operation="capable" class="cap" profile="unprivileged_userns" pid=5681 comm="chirpity" capability=21  capname="sys_admin"
[   60.851712] traps: chirpity[5671] trap int3 ip:5cbb804fa75a sp:7ffefc9ea600 error:0 in chirpity[5cbb7c9f9000+7e3a000]
```

 2. Next i tried *sudo ./chirpity-2.4.7.AppImage --no-sandbox* and a empty window showed up. ![grafik](https://github.com/user-attachments/assets/1ef7c546-bcb2-4cd1-99b1-145eff1a06c1)
 3. Your solution with *sudo systemctl ....* worked finally

Thanks for your help.

Best wishes Johannes

**Reply by Mattk70** [reply]
I'm glad that worked, by the way, I have [AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher) on my system, which allows you to integrate Chirpity into the desktop launcher. This may also be the reason I don't get the problem you've been having.
